pages/1_orders_and_revenue.py

```diff
@@ -18,49 +18,163 @@
 
 st.divider()
 
-st.dataframe(data.head(10)) ## This is here just as an example. This can be deleted during development and before release.
+## Ensure all date columns are in datetime format
+date_columns = ['created_at', 'customer_created_at', 'payment_at', 'subscription_period_started_at', 'subscription_period_ended_at']
+for col in date_columns:
+    if col in data.columns:
+        data[col] = pd.to_datetime(data[col], errors='coerce')
 
 st.divider()
 
-# Everything below this point will be generated by the assigned analyst. 
-# The framework has been set, but no visualization or data processing has been applied.
-# Please perform any data processing in this file and not within the filter files. We can discuss upon completion if it makes sense to add any code to the filters file.
-
-## KPI Metrics which need to be updated.
+# KPI Metrics
 with st.container():
     col1, col2, col3, col4 = st.columns(4)
 
+    total_revenue = data['total_amount'].sum()
+    number_of_orders = data['header_id'].nunique()
+    number_of_customers = data['customer_id'].nunique()
+    new_customers = data[data['customer_created_at'] == data['created_at']]['customer_id'].nunique()
+
     with col1:
-        st.metric(label="**Total Revenue**", value="TBD")
+        st.metric(label="**Total Revenue**", value=f"${total_revenue:,.2f}")
 
     with col2:
-        st.metric(label="**Number of Orders**", value="TBD")
+        st.metric(label="**Number of Orders**", value=number_of_orders)
 
     with col3:
-        st.metric(label="**Number of Customers**", value="TBD")
+        st.metric(label="**Number of Customers**", value=number_of_customers)
 
     with col4:
-        st.metric(label="**New Customers**", value="TBD")
+        st.metric(label="**New Customers**", value=new_customers)
+
+
+# Time series charts
+data['month'] = data['created_at'].dt.to_period('M').dt.to_timestamp()
 
-# Time series charts need to be built out
 with st.container():
     row1_col1, row1_col2 = st.columns(2)
     row2_col1, row2_col2 = st.columns(2)
 
     with row1_col1:
         st.markdown("**Total Revenue Over Time**")
+        revenue_over_time = data.groupby('month')['total_amount'].sum().reset_index()
+        fig = px.line(revenue_over_time, x='month', y='total_amount')
+        fig.update_yaxes(tickprefix='$', tickformat='~s', title_text='Revenue ($1000s)')
+        fig.update_xaxes(title_text='Month')
+        st.plotly_chart(fig, use_container_width=True)
```

Comment:
I noticed you used a line graph for this visualization. However this tells a misleading story. When looking at the chart the March 2024 month looks like it is nearly at 0. However, it is actually at 680k. I find it misleading that our 0 gridpoint is actually 680k. Can we change this to be a bar chart and show the total revenue from 0 up each month.

Reply:
Same comment for the Total number of orders and new customers charts

Reply:
@fivetran-reneeli it doesn't explicitly start from 0, but when looking at the chart it looks likeJuly is all the way at the bottom (visual representation of 0). If we make this a bar chart then we can highlight the entirety of the month and not just the change month over month.